src/vs/workbench/api/common/extHostTypes.ts

```diff
@@ -2745,7 +2745,7 @@ export class Breakpoint {
 	readonly hitCondition?: string;
 	readonly logMessage?: string;
 
-	protected constructor(enabled?: boolean, condition?: string, hitCondition?: string, logMessage?: string) {
+	protected constructor(enabled?: boolean, condition?: string, hitCondition?: string, logMessage?: string, id?: string) {
```

Comment:
@jrieken this is an extension API object that is constructed by extensions and also by us. I just want to enable us to construct it with an id internally, but not allow extensions to. Is it ok to add this constructor parameter here and just leave it out of the signature in vscode.d.ts? Even though they could technically still pass an id, it would be misusing the API.

Reply:
> Even though they could technically still pass an id, it would be misusing the API.

Yeah, there is the API has described by `vscode.d.ts` and there is the API discoverable by debugging and introspection... Ideally, the API contract is true down to the implementation. In this particular case you have a breakpoint class with the full constructor and a sub-type for the API that doesn't have the ctor-arg

Reply:
I didn't want to use subclasses because I think extensions use `instanceof` to disambiguate breakpoint kinds.